dns_test.go

```diff
@@ -374,17 +374,47 @@ func TestDNSProtocol(t *testing.T) {
 
 		_, port, _ := net.SplitHostPort(addr.String())
 
-		// Force IPv4
+		// Default DNS Server port
 		module := Module{
 			Timeout: time.Second,
 			DNS: DNSProbe{
 				QueryName: "example.com",
-				Protocol:  protocol + "4",
+				Protocol:  protocol,
 			},
 		}
 		recorder := httptest.NewRecorder()
 		registry := prometheus.NewRegistry()
-		result := probeDNS(net.JoinHostPort("localhost", port), recorder, module, registry)
+		result := probeDNS("localhost", recorder, module, registry)
+		if result {
+			t.Fatalf("DNS protocol: \"%v\" with localhost:53 succeeded, expected failure.", protocol)
```

Comment:
Yes, the test server almost would not start up at `localhost:53`, due to [test server](https://github.com/Colstuwjx/blackbox_exporter/blob/5fc97cca1dae3824b1a598b122d11a7bdd38d891/dns_test.go#L37) listen to a random port `:0`. So I designed that case to a `failure`.

Is there another pretty way to test it? I've tried `startDNSServer` with 53 port, but it didn't work...
Maybe [add default port support](https://github.com/miekg/dns/issues/491) with the upstream go dns would be better, but I'm not sure whether upstream will accept this dirty judgement, as the upstream repo is using `addr` at all place.